src/broker/shard_proxy.c

```diff
@@ -42,6 +42,7 @@
 /* SHARD SHM */
 int appl_server_shm_id = -1;
 T_SHM_APPL_SERVER *shm_as_p = NULL;
+T_SHM_APPL_SERVER *shm_appl = NULL;
```

Comment:
Why was shm_appl created additionally instead of using shm_as_p?
In your modified code, the shm_appl variable is only used in the set_net_buf_size() function.

Reply:
cas & cascgw uses **shm_appl** for shared memory reference, it is defined in cas.c.
I try to change/set netbuf size in car_net_buf.c using shared memory value **shm_appl->net_buf_size**,
it will be set in broker_config ().

However, shard proxy defined **shm_as_p** instead of **shm_appl**, moreover, it do not uses NET_BUF_SIZE.
If shm_appl is not defined, a **syntax error occurs when compiling shard proxy**.
If net_buf_init () or set_net_buf_size () in shard proxy, value of shm_appl will be NULL and **_net_buf_size** will be set to 16K.

Reply:
OK, I got it.